beta-operator-rewards/rewards-requirements.ts

```diff
@@ -0,0 +1,229 @@
+import { Contract } from "ethers";
+import { program } from "commander";
+import * as fs from "fs";
+import { ethers } from "ethers";
+import {
+  abi as RandomBeaconABI,
+  address as RandomBeaconAddress,
+} from "@keep-network/random-beacon/artifacts/RandomBeacon.json";
+import {
+  abi as WalletRegistryABI,
+  address as WalletRegistryAddress,
+} from "@keep-network/ecdsa/artifacts/WalletRegistry.json";
+import axios from "axios";
+
+export async function calculateBetaOperatorFactors() {
+  program
+    .version("0.0.1")
+    .requiredOption(
+      "-s, --start <timestamp>",
+      "starting time for rewards calculation"
+    )
+    .requiredOption(
+      "-e, --end <timestamp>",
+      "ending time for rewards calculation"
+    )
+    .requiredOption("-i, --interval <timestamp>", "scrape interval") // IMPORTANT, it has to be the same as in prometheous config
+    .requiredOption(
+      "-a, --api <prometheus api>",
+      "prometheus API"
+    )
+    .requiredOption(
+      "-j, --job <prometheus job>",
+      "prometheus job"
+    )
+    .requiredOption(
+      "-o, --output <file>",
+      "output JSON file"
+    )
+    .parse(process.argv);
+
+  const options = program.opts();
+
+  const queryStep = 600; // 10min in sec
+  const prometheous_job = options.job;
+  const prometheusAPI = options.api
+  const startRewardsTimestamp = parseInt(options.start);
+  const endRewardsTimestamp = parseInt(options.end);
+  const scrapeInterval = parseInt(options.interval);
+  const rewardsInterval = endRewardsTimestamp - startRewardsTimestamp;
+  const requiredUptime = 96; // percent
+  const preParamsAvgInterval = 24 // hours
+  const preParamsResolution = 24 // hours
+  const minPreParams = 500 // min requirement for pre params daily avg
+  const factors = {
```

Comment:
As far as I can tell, this `factors` object is never used as a collection, rather it's used to group some related constants. As in, later we might call: `peerData.set(factors.isBeaconAuthorized, 0);` or `peerData.set(factors.upTime, upFactor);`, and so having `factors.upTime` defined saves us from typing out `peerData.set("upTime", upFactor)`, which would be a magic string.

What do you think about taking this one step further, defining a file like `reward-constants.ts` and exporting a bunch of ALL_CAPS constants so their purpose is *super clear*?

Then, this file would be calling `peerData.set(UP_TIME, upFactor)`, and `if (Number(peerPreParams[1]) < MIN_PRE_PARAMS) {`